packages/dappmanager/src/calls/wifi.ts

```diff
@@ -0,0 +1,76 @@
+import { ComposeFileEditor } from "../modules/compose/editor";
+import { logContainer } from "../modules/docker/api";
+import { listContainer } from "../modules/docker/list";
+import { CurrentWifiCredentials, WifiReport } from "../types";
+import params from "../params";
+
+const WIFI_KEY_SSID = "SSID";
+const WIFI_KEY_PASSWORD = "WPA_PASSPHRASE";
+
+/**
+ * Get last log from wifi container
+ * Contains useful information for the user, check echos before exit codes:
+ * https://github.com/dappnode/DNP_WIFI/blob/master/build/wlanstart.sh
+ */
+async function getWifiLastLog(): Promise<string> {
+  return await logContainer(params.wifiContainerName, { tail: 1 });
```

Comment:
You can improve this function by checking the time the container stopped and requesting the logs from that time and now. Then look for the last log that matches some expected pattern.

Reply:
AFAIK in most of cases when a container stops, the last log is the error log explaining why the container stopped. In this specific use case, the wifi container exits at least three times with a meaningful log
https://github.com/dappnode/DNP_WIFI/blob/00685cf6160df4b8daeabd77246cb0af04a513f8/build/wlanstart.sh#L5
https://github.com/dappnode/DNP_WIFI/blob/00685cf6160df4b8daeabd77246cb0af04a513f8/build/wlanstart.sh#L34
https://github.com/dappnode/DNP_WIFI/blob/00685cf6160df4b8daeabd77246cb0af04a513f8/build/wlanstart.sh#L52

Also, I do not see the point of requesting the time of when the container stopped. After that time there will not be logs since the container is not running anymore

I could add parsing functions to interpret logs